test/e2e/tests/hardware-wallets/ledger/ledger-swap.spec.ts

```diff
@@ -75,9 +62,11 @@ describe('Ledger Swap', function () {
 
         await swapPage.enterSwapAmount('2');
         await swapPage.selectDestinationToken('DAI');
-        await swapPage.dismissManualTokenWarning();
 
+        await swapPage.dismissManualTokenWarning();
         await swapPage.checkSwapButtonIsEnabled();
+        // To mitigate flakiness where the Swap page is re-rendered after submitting the swap (#36501)
+        await driver.delay(5000);
```

Comment:
this fixes the issue: it's a bug on the application side, but until that's not fixed, we can have the hardcoded delay to mitigate the edge case

https://github.com/MetaMask/metamask-extension/issues/36501